packages/design-system/src/components/Enumeration/Enumeration.component.tsx

```diff
@@ -0,0 +1,173 @@
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { AutoSizer, InfiniteLoader, List } from 'react-virtualized';
+
+import { I18N_DOMAIN_DESIGN_SYSTEM } from '../constants';
+import { EmptyState } from '../EmptyState';
+import { StackHorizontal } from '../Stack';
+import { EnumerationMode, EnumerationProps, UiEnumerationItem } from './Enumeration.types';
+import { EnumerationHeader } from './EnumerationHeader/EnumerationHeader.component';
+import { EnumerationItem } from './EnumerationItem/EnumerationItem.component';
+
+import styles from './Enumeration.module.scss';
+
+const ENUMERATION_ITEM_HEIGHT = 38;
+
+export const Enumeration = ({
+	id,
+	items,
+	loadMoreRows,
+	onCreate,
+	onChange,
+	onEdit,
+	onImport,
+	onRemove,
+	title,
+}: EnumerationProps) => {
+	const { t } = useTranslation(I18N_DOMAIN_DESIGN_SYSTEM);
+
+	const [mode, setMode] = useState(EnumerationMode.VIEW);
+	const [selectedItems, setSelectedItems] = useState<string[]>([]);
+	const [filteredItems, setFilteredItems] = useState(items);
+	const [scrollToIndex, setScrollToIndex] = useState<number>();
+	const [uiItems, setUiItems] = useState<UiEnumerationItem[]>(
+		filteredItems.map(item => ({
+			value: item,
+			isToAnimate: false,
+		})),
+	);
+
+	const setIsToAnimate = (itemsToAnimate: string[], isToAnimate: boolean) => {
+		const newItems = [...uiItems];
+
+		itemsToAnimate.forEach(itemToAnimate => {
+			const indextToUpdate = newItems.findIndex(item => item.value === itemToAnimate);
+
+			if (indextToUpdate < 0) {
+				newItems.unshift({
+					value: itemToAnimate,
+					isToAnimate,
+				});
+			} else {
+				newItems[indextToUpdate].isToAnimate = isToAnimate;
+			}
+		});
+
+		setUiItems(newItems);
+	};
+
+	const onAnimate = (newItems: string[]) => {
+		setScrollToIndex(0);
+		setIsToAnimate(newItems, true);
+		setTimeout(() => {
+			setIsToAnimate(newItems, false);
+			setScrollToIndex(undefined);
+		}, 2500);
+	};
+
+	const handleOnCreate = async (newItem: string) => {
+		try {
+			if (newItem) {
+				await onCreate?.(newItem);
+				onChange([newItem, ...items]);
+				onAnimate([newItem]);
+			}
+		} catch (e) {
+			//The parent component must do the error handling
```

Comment:
How the parent know there is an error to handle ? Here the empty catch will just silence the error 🤔

Reply:
The point was on each callback (onCreate, onEdit, onDelete) the parent would have to deal with any possible error handling. The try catch is just to prevent the app crashing. What do you think? 🤔

Reply:
Ok, got it 👍